### PR TITLE
[New] Add videojs-vast-plugin@0.2.1 w/ git auto-update

### LINKED ISSUE #9820: [Request] Add videojs-vast-plugin
**Library name:** videojs-vast-plugin
**Git repository url:** https://github.com/theonion/videojs-vast-plugin
**License(s):** MIT
**Official homepage:** https://theonion.github.io/videojs-vast-plugin/

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀 .

If you'll like to work locally, I'll suggest you start pulling cdnjs repository with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are so many files in this repo, without sparseCheckout, the whole process will be slow and will use more than 80GB disk space, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄

### PR DESCRIPTION
**Pull request by sashberd**
close #9820, cc @theonion, using web-minify-helper tool for minifing

Pull request for issue: #9820 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/theonion/videojs-vast-plugin/tree/master
 * Official website (optional, not the repository): http://theonion.github.io/videojs-vast-plugin/
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 34
   - Count of stars: 215
   - Count of forks: 134
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sashberd** [comment]
@maruilian11 video.ads.js is file from different library that can be found here
https://github.com/videojs/videojs-contrib-ads

**Comment by maruilian11** [comment]
@sashberd Thanks! Then just add `videojs.vast.css`. Would you like to open a PR for `videojs-contrib-ads`? Thank you!

**Comment by sashberd** [comment]
I think we already have it inside
image
I will update css as soon as possible

**Comment by maruilian11** [comment]
@sashberd Thanks. I searched for `videojs.ads.css` and didn't get the result. I will remember to search for its project name next time.

File: ajax/libs/videojs-vast-plugin/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      {
+        "basePath": "",
+        "files": [
+          "videojs.vast.*"
+        ]
+      },
+      {
+        "basePath": "lib",
+        "files": [
+          "vast-client.js"
```
Comment (maruilian11) [inline ajax/libs/videojs-vast-plugin/package.json]: Let's use `vast-client*` to fetch minified file and map file in the future. Thanks.

**Comment by sashberd** [comment]
@maruilian11 Changes were updated

**Comment by maruilian11** [comment]
@sashberd Did you cut and paste the code on the website to minify? Or did you use `command`?(`$minifier->minify('/target/path.js');`) We hope you can use the command to minify it to prevent any human mistake. Thanks!

**Comment by sashberd** [comment]
@maruilian11 It works via php Composer, I do not have PHP on my computer. I will glad to use PeterDaveHello minifier, but it is not working on my machine, thus I am using that I have with copy/paste

**Comment by maruilian11** [comment]
@sashberd Have you tried to use uglify-js in windows? 
https://github.com/mishoo/UglifyJS2
https://simpletutorials.com/c/1875/How%20to%20Minify%20Your%20JavaScript%20Using%20UglifyJS2%20(Windows)

**Comment by sashberd** [comment]
@maruilian11 The PR was updated. I used koala tool that uses UglifyJS in its algorithm

File: ajax/libs/videojs-vast-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "target": "git://github.com/theonion/videojs-vast-plugin.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "videojs.vast.*"
+        ]
+      },
+      {
+        "basePath": "lib",
```
Comment (extend1994) [inline ajax/libs/videojs-vast-plugin/package.json]: The files under `lib` belongs other projects: [videojs-contrib-ads](https://github.com/videojs/videojs-contrib-ads) & [vast-client-js](https://github.com/dailymotion/vast-client-js).
The former one had been added, which can be found here https://cdnjs.com/libraries/videojs-contrib-ads.
The latter one should be added independently (Open a issue or directly open a pull request and cc this pull request). I will help open a issue for this in a minute this time.

So please remove L27 to L32. Thank you.

**Comment by sashberd** [comment]
@extend1994 The PR was updated FYR

**Comment by sashberd** [comment]
@maruilian11 @extend1994 ping

**Comment by sashberd** [comment]
@extend1994 The PR message was updated. FYR

File: ajax/libs/videojs-vast-plugin/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ],
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/theonion/videojs-vast-plugin.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "videojs.vast.*"
```
Comment (extend1994) [inline ajax/libs/videojs-vast-plugin/package.json]: `videojs.vast*` seems better?

Comment (sashberd) [reply ajax/libs/videojs-vast-plugin/package.json]: Agree

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Pull request merged**